aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/TransferTable.kt

```diff
@@ -219,6 +222,9 @@ internal class TransferTable {
             if (TABLE_VERSION_9 in (oldVersion + 1)..newVersion) {
                 addVersion9Columns(database)
             }
+            if (TABLE_VERSION_10 in (oldVersion + 1)..newVersion) {
+                addVersion10Columns(database)
```

Comment:
This is good for the update path, but the create code is missing the new region column

Reply:
if the the new column in the create code, we'll get a column already exists error.  The database gets created at version 1 and then upgrades to 10. 
https://github.com/aws-amplify/amplify-android/pull/2899/files#diff-26d789d324d49670cba8878a5cdd9b2eca0f1ea1ad29f684c46d5a2c8cd5e14dR186-R187

Reply:
Yikes, I see this now. This is NOT how it is supposed to work. Lets huddle on this.

Reply:
TransferDBHelper extends SQLiteOpenHelper which is a SQLite helper class that Android provides.

onCreate, onUpdate, etc are supposed to be called by Android, not manually called.

Looks like we have `TransferTable` to then route these calls to. Right now, TransferDbHelper sets the version of the database, not `TransferTable`.

So our `TransferTable's create statement is always creating v1, and then running through all upgrades to arrive at how VERSION 10 should actually look.

While efficiency would take a small hit (probably negligible in this case), there's no reason we should be calling onUpgrade from within onCreate. This is not how DB creation/migrations are supposed to work and could result in us making errors in the future.

The correct think to do would be to maintain a single create statement and then proper onUpgrade/onDowngrade sqlite statements. Looks like this is nothing introduced by your changes, and is how the db is written to behave in other areas of our code as well.

Reply:
I've added code comments to clarify the current design and prevent future updates to the create sql statement.